packages/fuels-programs/src/script_calls.rs

```diff
@@ -246,3 +314,32 @@ where
         ))
     }
 }
+
+#[async_trait::async_trait]
+impl<T, D> TxDependencyEstimation for ScriptCallHandler<T, D>
+where
+    T: Account,
+    D: Tokenizable + Parameterize + Debug + Send + Sync,
+{
+    async fn simulate(&mut self) -> Result<()> {
+        self.simulate().await?;
+
+        Ok(())
+    }
+
+    fn append_missing_deps(mut self, receipts: &[Receipt]) -> Self {
+        if is_missing_output_variables(receipts) {
+            self = self.append_variable_outputs(1)
+        }
+        if let Some(panic_receipt) = find_contract_not_in_inputs(receipts) {
+            let contract_id = Bech32ContractId::from(
+                *panic_receipt
+                    .contract_id()
+                    .expect("Panic receipt must contain contract id."),
+            );
+            self = self.append_contract(contract_id);
+        }
+
+        self
+    }
```

Comment:
why cannot this method have its default implementation inside the trait? It looks like it's the same for both `ScriptCall` and `ContractCall`, and can be overidden for `MultiCall`.